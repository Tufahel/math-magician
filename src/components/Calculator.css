.grid-container {
  display: grid;
  grid-template-columns: auto auto auto auto;
  gap: 2px;
  background-color: rgb(164, 164, 164);
  padding: 2px;
  margin: 10px;
}

.grid-container > div {
  background-color: rgba(255, 255, 255, 0.8);
  text-align: center;
  padding: 20px 0;
  font-size: 30px;
}

button {
  padding: 30px;
}

p {
  padding: 10px;
  font-size: 30px;
}

.item17 {
  grid-column-start: 1;
  grid-column-end: 3;
}

.item0 {
  grid-column-start: 1;
  grid-column-end: 5;
  background-color: rgb(164, 164, 164) !important;
  color: white;
  text-align-last: right;
  margin-right: 2px;
}

.item4,
.item8,
.item12,
.item16,
.item19 {
  background-color: rgb(255, 81, 0) !important;
  color: white;

