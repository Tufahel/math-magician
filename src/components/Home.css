
.home {
  text-align: left;
  margin: 40px 20px;
}

.home-title {
  margin-left: 10px;
  font-size: 30px;
}
