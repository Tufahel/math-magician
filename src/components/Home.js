import React from 'react';

function Home() {
  return (
    <div>
      <h3>Welcome to our page!</h3>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
        Quia tempore veritatis error ratione repellat qui, dolorum
        consequatur mollitia placeat tempora magnam commodi!
        Ipsa molestiae officiis, unde et porro nisi illo.
      </p>
      <br />
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit.
        Voluptate odio quasi veritatis enim non dolor harum sunt,
        quas ipsam laboriosam dicta repudiandae autem atque,
        architecto repellat obcaecati? Temporibus, delectus aperiam.
      </p>
    </div>
  );
}

export default Home;
