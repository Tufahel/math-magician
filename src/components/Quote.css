.quote-body {
  text-align: left;
  margin: 200px 30px;
}

.writer {
  font-weight: 700;
}
