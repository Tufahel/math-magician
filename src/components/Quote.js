import React from 'react';

function Quote() {
  return (
    <div>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
        Perspiciatis facilis repudiandae cupiditate veritatis nesciunt
        eius nemo sed nam. Quod, similique blanditiis culpa iure quisquam
        quo ullam nihil asperiores. Facere, quod.
      </p>
    </div>
  );
}

export default Quote;
