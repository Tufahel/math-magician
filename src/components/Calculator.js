import React, { useState } from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [calc, setCalc] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleCalculate = (event) => {
    setCalc((calc) => calculate(calc, event.target.textContent));
  };

  return (
    <div className="grid-container">
      <p className="item0">
        {calc.total}
        {' '}
        {calc.operation}
        {' '}
        {calc.next}
      </p>
      <button onClick={handleCalculate} type="button" className="item1">AC</button>
      <button onClick={handleCalculate} type="button" className="item2">+/-</button>
      <button onClick={handleCalculate} type="button" className="item3">%</button>
      <button onClick={handleCalculate} type="button" className="item4">รท</button>
      <button onClick={handleCalculate} type="button" className="item5">7</button>
      <button onClick={handleCalculate} type="button" className="item6">8</button>
      <button onClick={handleCalculate} type="button" className="item7">9</button>
      <button onClick={handleCalculate} type="button" className="item8">x</button>
      <button onClick={handleCalculate} type="button" className="item9">4</button>
      <button onClick={handleCalculate} type="button" className="item10">5</button>
      <button onClick={handleCalculate} type="button" className="item11">6</button>
      <button onClick={handleCalculate} type="button" className="item12">-</button>
      <button onClick={handleCalculate} type="button" className="item13">1</button>
      <button onClick={handleCalculate} type="button" className="item14">2</button>
      <button onClick={handleCalculate} type="button" className="item15">3</button>
      <button onClick={handleCalculate} type="button" className="item16">+</button>
      <button onClick={handleCalculate} type="button" className="item17">0</button>
      <button onClick={handleCalculate} type="button" className="item18">.</button>
      <button onClick={handleCalculate} type="button" className="item19">=</button>
    </div>
  );
};

export default Calculator;
