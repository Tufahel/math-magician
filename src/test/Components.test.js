import {
  render,
} from '@testing-library/react';
import Calculator from '../components/Calculator';
import Home from '../components/Home';
import Quote from '../components/Quote';

describe('test snapshot for components', () => {
  test('Create a Home snapshot', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment(<Home />)).toMatchSnapshot();
  });
  test('Create a Caculator snapshot', () => {
    const { asFragment } = render(<Calculator />);
    expect(asFragment(<Calculator />)).toMatchSnapshot();
  });
  test('Create a Quote snapshot', () => {
    const { asFragment } = render(<Quote />);
    expect(asFragment(<Quote />)).toMatchSnapshot();
  });
});
