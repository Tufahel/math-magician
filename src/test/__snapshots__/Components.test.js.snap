// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot for components Create a Caculator snapshot 1`] = `
<DocumentFragment>
  <div
    class="body"
  >
    <h2>
      Lets Do Some Math!
    </h2>
    <div
      class="grid-container"
    >
      <p
        class="item0"
      >
          
      </p>
      <button
        class="item1 item"
        type="button"
      >
        AC
      </button>
      <button
        class="item2 item"
        type="button"
      >
        +/-
      </button>
      <button
        class="item3 item"
        type="button"
      >
        %
      </button>
      <button
        class="item4 item"
        type="button"
      >
        ÷
      </button>
      <button
        class="item5 item"
        type="button"
      >
        7
      </button>
      <button
        class="item6 item"
        type="button"
      >
        8
      </button>
      <button
        class="item7 item"
        type="button"
      >
        9
      </button>
      <button
        class="item8 item"
        type="button"
      >
        x
      </button>
      <button
        class="item9 item"
        type="button"
      >
        4
      </button>
      <button
        class="item10 item"
        type="button"
      >
        5
      </button>
      <button
        class="item11 item"
        type="button"
      >
        6
      </button>
      <button
        class="item12 item"
        type="button"
      >
        -
      </button>
      <button
        class="item13 item"
        type="button"
      >
        1
      </button>
      <button
        class="item14 item"
        type="button"
      >
        2
      </button>
      <button
        class="item15 item"
        type="button"
      >
        3
      </button>
      <button
        class="item16 item"
        type="button"
      >
        +
      </button>
      <button
        class="item17 item"
        type="button"
      >
        0
      </button>
      <button
        class="item18 item"
        type="button"
      >
        .
      </button>
      <button
        class="item19 item"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`test snapshot for components Create a Home snapshot 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <h3
      class="home-title"
    >
      Welcome to our page!
    </h3>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Quia tempore veritatis error ratione repellat qui, dolorum consequatur mollitia placeat tempora magnam commodi! Ipsa molestiae officiis, unde et porro nisi illo.
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Voluptate odio quasi veritatis enim non dolor harum sunt, quas ipsam laboriosam dicta repudiandae autem atque, architecto repellat obcaecati? Temporibus, delectus aperiam.
    </p>
  </div>
</DocumentFragment>
`;

exports[`test snapshot for components Create a Home snapshot 2`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <h3
      class="home-title"
    >
      Welcome to our page!
    </h3>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Quia tempore veritatis error ratione repellat qui, dolorum consequatur mollitia placeat tempora magnam commodi! Ipsa molestiae officiis, unde et porro nisi illo.
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Voluptate odio quasi veritatis enim non dolor harum sunt, quas ipsam laboriosam dicta repudiandae autem atque, architecto repellat obcaecati? Temporibus, delectus aperiam.
    </p>
  </div>
</DocumentFragment>
`;

exports[`test snapshot for components Create a Quote snapshot 1`] = `
<DocumentFragment>
  <div
    class="quote-body"
  >
    <p>
      "As far as the laws of mathematics refer to reality, they are not certain, and as far as they are certain, they do not refer to reality."
      <br />
      <br />
      <span
        class="writer"
      >
        — Albert Einstein
      </span>
    </p>
  </div>
</DocumentFragment>
`;
